src/provider.tsx

```diff
@@ -64,12 +64,28 @@ class LDProvider extends React.Component<ProviderConfig, HocState> implements En
     });
   };
 
-  async componentDidMount() {
-    const { clientSideID, user, flags, options } = this.props;
+  initLDClient = async () => {
+    const { clientSideID, flags, options, user } = this.props;
     const reactOptions = this.getReactOptions();
     const { flags: fetchedFlags, ldClient } = await initLDClient(clientSideID, user, reactOptions, options, flags);
     this.setState({ flags: fetchedFlags, ldClient });
     this.subscribeToChanges(ldClient);
+  };
+
+  componentDidMount() {
+    const { user, waitForUserToInitializeClient } = this.props;
+    if (waitForUserToInitializeClient && !user) {
+      return;
+    }
+    this.initLDClient();
+  }
+
+  componentDidUpdate(prevProps: ProviderConfig) {
+    const { user, waitForUserToInitializeClient } = this.props;
+    const userJustLoaded = !prevProps.user && user;
+    if (waitForUserToInitializeClient && userJustLoaded) {
+      this.initLDClient();
+    }
```

Comment:
Are you able to write tests for these please? Really appreciate your help, thanks.